packages/go_router/lib/src/configuration.dart

```diff
@@ -0,0 +1,149 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'configuration.dart';
+import 'logging.dart';
+import 'path_utils.dart';
+import 'typedefs.dart';
+
+export 'route.dart';
+export 'state.dart';
+
+/// The route configuration for GoRouter, configured by the app
+class RouteConfiguration {
+  /// Constructs a RouterConfiguration
+  RouteConfiguration({
+    required this.routes,
+    required this.redirectLimit,
+    required this.topRedirect,
+  }) {
+    _cacheNameToPath('', routes);
+    assert(() {
+      _debugKnownRoutes();
+      return true;
+    }());
+  }
+
+  /// List of top level routes used by the go router delegate
+  final List<GoRoute> routes;
+
+  /// The limit for the number of consecutive redirects
+  final int redirectLimit;
+
+  /// Top level page redirect.
+  final GoRouterRedirect topRedirect;
+
+  final Map<String, String> _nameToPath = <String, String>{};
+
+  /// Throws a [RouteConfigurationError] if this configuration is invalid
+  void validate() {
```

Comment:
Ideally one shouldn't be able to create an invalid `RouteConfiguration` instance. Are there any scenarios where you'd want to create a `RouteConfiguration` without running `validate`? If no, what do you think of making this a factory constructor?

Reply:
Ah the parser used to enforce this validation in its constructor:

https://github.com/flutter/packages/blob/a637b4a038c69b079d55d5b8e07bc823a279e162/packages/go_router/lib/src/go_route_information_parser.dart#L32-L39

Reply:
I agree, how about we throw an error in the `RouteConfiguration` constructor? https://github.com/flutter/packages/pull/2317/commits/db91f560bcc39434a5320b8b3e6a98c6f2c83ad1